packages/eas-cli/src/submit/context.ts

```diff
@@ -41,15 +41,15 @@ export async function createSubmissionContextAsync<T extends Platform>(params: {
   platform: T;
   profile: SubmitProfile<T>;
   projectDir: string;
-  projectId: string;
   applicationIdentifier?: string;
   actor: Actor;
+  getDynamicProjectConfigAsync: DynamicConfigContextFn;
 }): Promise<SubmissionContext<T>> {
   const { applicationIdentifier, projectDir, nonInteractive, actor } = params;
-  const exp = getExpoConfig(projectDir, { env: params.env });
+  const { exp, projectId } = await params.getDynamicProjectConfigAsync({ env: params.env });
```

Comment:
It might be a bit cleaner to pass exp and projected
- When this code is called from submit command you would run the same code as here
- When this code is run as a part of the build command with auto-submit option you already have those value in buildCtx

Reply:
Good suggestion! Will do in a follow-up: https://linear.app/expo/issue/ENG-6471/clean-up-getdynamicprojectconfigasync-in-submit-context